app/src/main/java/it/niedermann/nextcloud/deck/remote/api/IResponseCallback.java

```diff
@@ -13,6 +14,9 @@ default void onError(Throwable throwable) {
         DeckLog.logError(throwable);
     }
 
+    default void onResponseWithHeaders(T response, Headers headers) {
+        onResponse(response);
```

Comment:
Currently `onResponse` is the default implementation and `onResponseWithHeaders` is the optional implementation that calls `onResponse`.

While most of the consumers won't need headers, having both methods can lead to confusion: The current two consumers do implement only `onResponseWithHeaders`. This leads to two issues:

* As consumer of this interface it is not clear whether both methods are called and if yes, in which order and sequentially or parallel?
* As supplier it is also unclear whether both methods must be called or only one. If only `onResponseWithHeaders` is called by the supplier (as in the current two implementations), can we trust that the consumers call `super` or do not put dead code in `onResponse`?

I see two possible solutions:

1. Drop `onResponse` completely in favour of `onResponseWithHeaders` (and rename it to `onResponse` for convenience).
  Pro: One API to rule them all, no confusion about call hierarchy
  Contra: Most consumers won't need the second argument, but does it hurt?
2. Add `@CallSuper` annotation to `onResponseWithHeaders` to "enforce" feeding both methods.
  Pro: Most consumers won't need an implementation with two arguments
  Contra: Each consumer has to think about when which method gets called, the confusion stays, even if it is more "proof". `@CallSuper` is not a JVM level check, it's just an IDE hint that can be ignored.

I am ready to make the necessary changes in case you agree.

Reply:
I changed tat according to solution 1. Was actually more annoying then I initially thought, but here you go.